packages/context/src/binding.ts

```diff
@@ -282,6 +303,32 @@ export class Binding {
     return this;
   }
 
+  /**
+   * Set options for the binding
+   * @param options Options object or a context object that binds all options
+   */
+  withOptions(options: {[name: string]: BoundValue} | Context): this {
+    if (isPromise(options)) {
+      throw new Error(
+        'Promises are not allowed for options. Use a plain object or ' +
+          'an instance of Context to support asynchronous resolutions.',
+      );
+    }
+    if (!this.options) this.options = new Context();
```

Comment:
IIUC, this means that options entries cannot reference any dependencies from the application context, because the options context is not part of the application-request context hierarchy. Is it a good thing?

Reply:
With the multi-parent ability, we can now resolve options with the composite context.